toolchain/supportconfig.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Basheer Khadarsabgari <basheer@linux.vnet.ibm.com>
+
+import os
+import re
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Supportconfig(Test):
+    """
+    The supportconfig script gathers system troubleshooting information on
+    SUSE Linux Enterprise Server systems.It captures the current system
+    environment and generates a tar-archive.
+
+    The script file collects complementary information to the dbginfo.sh
+    script. The supportconfig script is part of the Supportutils package.
+    """
+
+    def setUp(self):
+        """
+        Install all the dependency packages required for Test.
+        """
+        sm = SoftwareManager()
+        dist = distro.detect()
+
+        if "SuSE" not in dist.name:
+            self.skip("supportconfig is supported on the SuSE Distro only.")
+
+        package = "supportutils"
+        if not sm.check_installed(package) and not sm.install(package):
```

Comment:
please here as well 

`if not sm.check_installed("supportutils") and not sm.install("supportutils"):`